ha/core/controllers/pcs/service_controller.py

```diff
@@ -52,20 +56,33 @@ def start(self, service: str, nodeids: list = None) -> dict:
         raise HAUnimplemented("This operation is not implemented.")
 
     @controller_error_handler
-    def stop(self, service: str, nodeids: list = None) -> dict:
+    def stop(self, nodeid: str, excludeResourceList: list = None) -> dict:
         """
         Stop service.
 
         Args:
-            service (str): Service name.
-            nodeids (list, optional): Node ids, if none then all node status.
-                    Defaults to None.
+            nodeid (str): Private fqdn define in conf store.
+            excludeResourceList (list): Service list which are not stopped.
 
         Returns:
             ([dict]): Return dictionary. {"status": "", "msg":""}
                 status: Succeeded, Failed, InProgress
         """
-        raise HAUnimplemented("This operation is not implemented.")
+        try:
+            resources: list = []
+            output, _, _ = self._execute.run_cmd(const.LIST_PCS_RESOURCES, check_error=False)
+            if "NO resources" not in output:
+                for resource in output.split("\n"):
+                    res = resource.split(":")[0]
+                    if res != "" and res not in resources and res not in excludeResourceList:
+                        resources.append(res)
+            for resource in resources:
+                self._execute.run_cmd(f"pcs resource ban {resource} {nodeid}")
+            Log.info(f"Waiting to stop resource on node {nodeid}")
+            time.sleep(const.BASE_WAIT_TIME)
```

Comment:
If you want to wait till resources have stopped, please add a check after sleep to verify the same. 
Otherwise, why is this sleep needed ?

Reply:
@ajaysrivas , do we need to wait for resources to be stopped here ?

Reply:
That's correct. We should have code to check if the resources are stopped. For now, add a TODO and move ahead. This check will be added when we add code for service management.

Reply:
yes returning SUCCEEDED is misleading 
we should check if service is stop 
we have status for service if not big change we can add status code

Reply:
status may need some time to consider all cases 
```
# crm_resource --resource  sspl-ll --locate srvnode-2
resource sspl-ll is running on: srvnode-1
resource sspl-ll is running on: srvnode-2
resource sspl-ll is running on: srvnode-3

# crm_resource --resource  health_monitor --locate srvnode-2
resource health_monitor is NOT running


```

Reply:
For now changed status from SUCCEEDED to INPROGRESS. added TODO to check resources are stopped.

Reply:
Please file a jira for this change and update that ticket id here

Reply:
Updated Jira_id in comments

Reply:
```
# TODO: Check if the resources are stopped EOS-23386
```
Looks good